src/templates/generate-config.py

```diff
@@ -31,6 +31,27 @@ def is_valid(test_harness_path):
       return False 
   return True 
 
+def extends_config(application_path): 
+  """
+  Checks if application under testing provides optional custom circleci yaml extension `circle_config.yml`. 
+  """
+  if not os.path.isdir(application_path):
+        return False
+
+  elements = [os.path.basename(name) for name in os.listdir(application_path)]
+  has_config = ('circle_config.yml' in elements)
+  return has_config 
+
+def load_circle_config(application_path):
+    """
+    Loads custom CircleCI configuration from 'circle_config.yml' if available.
+    """
+    config_path = os.path.join(application_path, 'circle_config.yml')
+    if os.path.isfile(config_path):
+        with open(config_path, 'r') as file:
+            return yaml.safe_load(file)
+    return None
```

Comment:
Is the user informed when the file is present but in an incorrect format? Can we add any validation here to inform the user about issues with additional steps?
we could implement something like this:
 ```python 
 schema = { ... }
 data = yaml.safe_load
 try:
            jsonschema.validate(instance=data, schema=schema)
 catch validation error
...
 ```

Reply:
That's great remark, I didn't add json validation as it'd require installing extra package. However, I've added in code validation for before and after formats. Let me know if that will work for you,